pkgs/applications/networking/cluster/hadoop/default.nix

```diff
@@ -124,38 +215,74 @@ let
-    opensslPkg = openssl_1_0_2;
+    version             = "2.7.7";
+    sources-sha256      = "1ahv67f3lwak3kbjvnk1gncq56z6dksbajj872iqd0awdsj3p5rf";
+    dependencies-sha256 = "1719jai4az4iyr4a2nz9b6pmb8ql2bbpyp40vsnhs6nwx3wsdf9c";
+    tomcat              = tomcat_6_0_48;
+    opensslPkg          = openssl_1_0_2;
+    protobufPkg         = protobuf2_5;
   };
   hadoop_2_8 = common {
-    version = "2.8.4";
-    sha256 = "16c3ljhrzibkjn3y1bmjxdgf0kn60l23ay5hqpp7vpbnqx52x68w";
-    dependencies-sha256 = "1j4f461487fydgr5978nnm245ksv4xbvskfr8pbmfhcyss6b7w03";
-    tomcat = tomcat_6_0_48;
-    opensslPkg = openssl_1_0_2;
+    version             = "2.8.5";
+    sources-sha256      = "0d1xbcdy5qgmh1gh5y8fv1gf0hpwgz3yzkpw6s0bc294bz2pzc35";
+    dependencies-sha256 = "0dr2i2j9qvnkqcras553s2m97q1v9ibkym8k47f7dj1xs0v04whr";
+    tomcat              = tomcat_6_0_48;
+    opensslPkg          = openssl_1_0_2;
+    protobufPkg         = protobuf2_5;
   };
   hadoop_2_9 = common {
-    version = "2.9.1";
-    sha256 = "0qgmpfbpv7f521fkjy5ldzdb4lwiblhs0hyl8qy041ws17y5x7d7";
-    dependencies-sha256 = "1d5i8jj5y746rrqb9lscycnd7acmxlkz64ydsiyqsh5cdqgy2x7x";
-    tomcat = tomcat_6_0_48;
-    opensslPkg = openssl_1_0_2;
+    version             = "2.9.2";
+    sources-sha256      = "1nifslax1pgzg6xsny7m5dyg3d8vqfpbzr0dmlxhbwm70pxw62jr";
+    dependencies-sha256 = "0r47p9mq7pwmma47rfh5in9fa3dc79zk7vp43fvcfgmbl08r54j2";
+    tomcat              = tomcat_6_0_48;
+    opensslPkg          = openssl_1_0_2;
+    protobufPkg         = protobuf2_5;
+    extraBuildInputs    = [ zstd ];
   };
-  hadoop_3_0 = common {
-    version = "3.0.3";
-    sha256 = "1vvkci0kx4b48dg0niifn2d3r4wwq8pb3c5z20wy8pqsqrqhlci5";
-    dependencies-sha256 = "1kzkna9ywacm2m1cirj9cyip66bgqjhid2xf9rrhq6g10lhr8j9m";
-    tomcat = null;
+  hadoop_2_10 = common {
+    version             = "2.10.1";
+    sources-sha256      = "04jmwjxq8hb52nmklswx3000hwnvi82pzmqwa5jkjyh31rbvq11w";
+    dependencies-sha256 = "1lhn9aqk9wf4fh08nk1cm5l8ic341lr4fc6f8c5vf3653aqc997h";
+    tomcat              = tomcat_8_5_43;
+    opensslPkg          = openssl_1_0_2;
+    protobufPkg         = protobuf2_5;
+    extraBuildInputs    = [ zstd ];
   };
   hadoop_3_1 = common {
-    version = "3.1.1";
-    sha256 = "04hhdbyd4x1hy0fpy537f8mi0864hww97zap29x7dk1smrffwabd";
-    dependencies-sha256 = "1q63jsxg3d31x0p8hvhpvbly2b07almyzsbhwphbczl3fhlqgiwn";
-    tomcat = null;
+    version             = "3.1.4";
+    sources-sha256      = "1syqfyd2f9zy6s7w5bhjkbz1xmrp63grw0mrn680n0lqkhj7kwzw";
+    dependencies-sha256 = "0bc90yw6q2g3y5qlsmvzv5ph5s8a1aqfn7iab2frv64mcqw71s5l";
+    opensslPkg          = openssl_1_1;
+    protobufPkg         = protobuf2_5;
+    extraBuildInputs    = [ zstd isa-l ];
+  };
+  hadoop_3_2 = common {
+    version             = "3.2.1";
+    sources-sha256      = "0903fpbgvx4fzccza9xagkw3ck2zbd0g2v9m337qbgs7q1bi7lrv";
+    dependencies-sha256 = "09rq1d3v0cwia8lllpb6jxdma27f3f90pnkh96f5aj7ahyds8r66";
```

Comment:
```
hash mismatch in fixed-output derivation '/nix/store/4a5nzp9nfd9xxkp6n4an8wb6vyjcil7d-hadoop-maven-deps-3.2.1':
  wanted: sha256:09rq1d3v0cwia8lllpb6jxdma27f3f90pnkh96f5aj7ahyds8r66
  got:    sha256:0f19k6zcfmk6027a3jd7350nldjgwiqvh2j1yb6smk6zml27j8wc
```
While testing with nixpkgs-review

Reply:
I am still getting `09rq1d3v0cwia8lllpb6jxdma27f3f90pnkh96f5aj7ahyds8r66`, so there must be another impurity than Maven range version.

Reply:
How do you want to proceed? Do we just merge this as is and fix later?

Reply:
No idea.
I thought about snapshotting the whole Maven and use immutable snapshots, but since `hadoop-3.3.0` (and `ozone-1.0.0`), there are also downloaded JS files from NPM which makes it more difficult.

Reply:
Also, version of Maven might affect the hash, because different Mavens download different plugins.